FindEGL add IMPORTED_NO_SONAME property

CoreELEC (official): 22.0-P_nightly_20240612 (Amlogic-ne.aarch64)
      Machine model: Dune/Homatics R 4K Plus
     CoreELEC dt-id: sc2_s905x4_sei_smb_280_id6
      Amlogic dt-id: sc2_s905x4_ah212-id6

CoreELEC:~ # /usr/lib/kodi/kodi.sh --standalone -fs &

/usr/lib/kodi/kodi.bin: error while loading shared libraries:
  /home/docker/build_dir/CoreELEC-22-ne/build.CoreELEC-Amlogic-ne.aarch64-22/toolchain/aarch64-libreelec-linux-gnu/sysroot/usr/lib/libEGL.so: cannot open shared object file: No such file or directory

--- a/cmake/modules/FindEGL.cmake    2024-01-15 18:58:20.000000000 +0100
+++ b/cmake/modules/FindEGL.cmake    2024-01-15 19:29:12.379513754 +0100
@@ -35,11 +35,16 @@ if(NOT TARGET ${APP_NAME_LC}::${CMAKE_FI
     check_include_files("EGL/egl.h;EGL/eglext.h;EGL/eglext_angle.h" HAVE_EGLEXTANGLE)
     unset(CMAKE_REQUIRED_INCLUDES)
 
-    add_library(${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME} UNKNOWN IMPORTED)
+    if(${EGL_LIBRARY} MATCHES ".+\.so$")
+      add_library(${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME} SHARED IMPORTED)
+    else()
+      add_library(${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME} UNKNOWN IMPORTED)
+    endif()
     set_target_properties(${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME} PROPERTIES
                                                                      IMPORTED_LOCATION "${EGL_LIBRARY}"
                                                                      INTERFACE_INCLUDE_DIRECTORIES "${EGL_INCLUDE_DIR}"
-                                                                     INTERFACE_COMPILE_DEFINITIONS HAS_EGL)
+                                                                     INTERFACE_COMPILE_DEFINITIONS HAS_EGL
+                                                                     IMPORTED_NO_SONAME TRUE)
 
     if(HAVE_EGLEXTANGLE)
       set_property(TARGET ${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME} APPEND PROPERTY
